### ISSUE TITLE
Remove bootstrap dependency...

### ISSUE THREAD

**Issue by bastienJS** [fraywing/textAngular#625]
The world is moving towards material design.

I do not want to use bootstrap for the text angular component.

I would like to use e.g. angularjs material design for all the toolbar buttons.

This is a feature request :-)

Should text-angular component not work together with every ui framework I want to plug in?

**Comment by SimeonC** [comment]
Yes that's true. And theoretically you can do almost all of this just by changing the classes, I've never tried it though.

**Comment by pbassut** [comment]
This is great. I think what textAngular depends on bootstrap to layout things properly could be done in a few lines with css. And also would be less likely to break the whole user layout, as it happened to me.

I'll try to come up with a PR for this.

**Comment by mwils-bp** [comment]
Hey, I started a version with Angular Material by editing the classes inside the ```textAngularSetup.js```, its not perfect but is defiantly do-able.

**Comment by bopm** [comment]
:+1:

**Comment by adriendomoison** [comment]
Did you find how to do it with the basic version of textAngular? In angular-material-textangular module, only the buttons are changed, I need to change the input style to material textarea, any idea?

**Comment by mwils-bp** [comment]
@adriendomoison, I have been meaning to get this updated, its all pure material now.

![angular-material-textangular](https://cloud.githubusercontent.com/assets/5865727/8262474/96c35eb6-16ca-11e5-8e4c-a18e1c802d70.png)

I'm planning on updating everything this weekend.

**Comment by adriendomoison** [comment]
@mwils-bp Whow ! That's impressive ! Very nice work, gratz, Do you plan to include this material textarea too ?
![material_textarea](https://cloud.githubusercontent.com/assets/9092210/8265242/2d28150c-16f5-11e5-9c89-5fb69f5d75d5.png)
(from https://material.angularjs.org/latest/#/demo/material.components.input)

I would love a Rich Toolbar that you plug to any textarea with a simple directive attribute, maybe I will do it myself if nobody integrate it before :)

**Comment by agjs** [comment]
I'm on this too and willing to assist if necessary. Bootstrap is really not interesting anymore and there are many other css frameworks out there. Being dependent on Bootstrap is just too lame nowdays.

**Comment by SimeonC** [comment]
Just to clarify, Bootstrap is not a dependancy - it's just the default styling. You can change the default classes we use in the textAngularSetup.js file which will allow you to use any other CSS based framework. With the exception of the popover which the css is included anyway.

As for material support, the most I'd suggest we do in textAngular itself is expose all the builder functions and allow them to be $decorated to allow people to implement it themselves.

**Comment by clementmas** [comment]
I'm also trying to get rid of Bootstrap to use Material Design Lite instead.

I don't think changing the `textAngularSetup.js` file is the way to go about doing this.
Overriding the defaults (https://github.com/fraywing/textAngular/wiki/Setting-Defaults) seems more appropriate.

Currently I can to something like :
`taTools.bold.iconclass = 'material-icons';`

But I would need to be able to do :
`taTools.bold.icontext = 'format_bold';`

Unfortunately this is not supported but it would be a great new feature.
Otherwise it means I have to register new tools using the 'display' attribute to show buttons with basically the same actions.

**Comment by shuhanxu** [comment]
+1 for this

**Comment by Mkuhneu** [comment]
Please make texAngular more themeable. I do not like required ui dependencies

**Comment by ferc** [comment]
+1

**Comment by MadJlzz** [comment]
+1

**Comment by pbassut** [comment]
For future readers, take a look at @SimeonC's comment here: https://github.com/fraywing/textAngular/issues/625#issuecomment-138380833

Bootstrap is not a dependency! Just the default styling. TextAngular can be easily customized to work with another theme just by changing the button classes.

**Comment by lewiswharf** [comment]
It is possible to use material design icons. Here's an example of how to replace the bold button with material design (I'm utilizing ngMaterial in terms of element structure).


    $provide.decorator('taTools', ['$delegate', function(taTools){
        taTools.bold.display = '<md-button name="bold" class="md-icon-button" aria-label="Format bold"><md-icon class="material-icons format_bold">format_bold</md-icon></md-button>';
        return taTools;
    }]);

I hope this helps!
